apis/postgresql/v1alpha1/extension_types.go

```diff
@@ -0,0 +1,115 @@
+/*
+Copyright 2020 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"context"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+	"github.com/crossplane/crossplane-runtime/pkg/reference"
+	"github.com/pkg/errors"
+)
+
+// ExtensionParameters are the configurable fields of a Extension.
+type ExtensionParameters struct {
+	// Extension name to be installed
+	Extension string `json:"extension"`
+
+	// Extension Version to be installed
+	// +immutable
+	// +optional
+	Version *string `json:"version,omitempty"`
+
+	// Schema for extension install.
+	// +optional
+	Schema *string `json:"schema,omitempty"`
+
+	// Database for extension install.
+	// +optional
+	Database *string `json:"database,omitempty"`
+
+	// DatabaseRef references the database object this extension it for.
+	// +immutable
+	// +optional
+	DatabaseRef *xpv1.Reference `json:"databaseRef,omitempty"`
+
+	// DatabaseSelector selects a reference to a Database this extension is for.
+	// +immutable
+	// +optional
+	DatabaseSelector *xpv1.Selector `json:"databaseSelector,omitempty"`
+}
+
+// A ExtensionSpec defines the desired state of a Extension.
```

Comment:
```suggestion
// ExtensionSpec defines the desired state of an Extension.
```

Super nitpicky but "an Extension" reads cleaner to me than "a Extension", can we use that (or just "Extension" on its own)?